core/src/main/java/org/bitcoinj/core/TransactionInput.java

```diff
@@ -359,15 +359,29 @@ public void connect(TransactionOutput out) {
      * @return true if the disconnection took place, false if it was not connected.
      */
     public boolean disconnect() {
-        if (outpoint.fromTx == null) return false;
-        TransactionOutput output = outpoint.fromTx.getOutput((int) outpoint.getIndex());
-        if (output.getSpentBy() == this) {
-            output.markAsUnspent();
+        if (outpoint.fromTx != null) {
+            TransactionOutput output = outpoint.fromTx.getOutput((int) outpoint.getIndex());
             outpoint.fromTx = null;
-            return true;
+            if (output.getSpentBy() == this) {
+                output.markAsUnspent();
+                return true;
+            } else {
+                return false;
+            }
         } else {
```

Comment:
This could be an else if, right? So the whole outer if would be if { } else if { } else { }. Maybe add some comments to the first two main branches of the if?